pkg/controller/plan/validation.go

```diff
@@ -112,35 +115,62 @@ func (r *Reconciler) validateVM(provider *api.Provider, plan *api.Plan) error {
 		Status:   True,
 		Reason:   NotFound,
 		Category: Critical,
-		Message:  "The VMs (list) contains invalid VMs.",
+		Message:  "VM not found.",
 		Items:    []string{},
 	}
 	notUnique := libcnd.Condition{
 		Type:     DuplicateVM,
 		Status:   True,
 		Reason:   NotUnique,
 		Category: Critical,
-		Message:  "The VMs (list) contains duplicate VMs.",
+		Message:  "Duplicate VM.",
 		Items:    []string{},
 	}
+	ambiguous := libcnd.Condition{
+		Type:     DuplicateVM,
+		Status:   True,
+		Reason:   Ambiguous,
+		Category: Critical,
+		Message:  "VM reference is ambiguous.",
+		Items:    []string{},
+	}
+	setOf := map[string]bool{}
+	hash := func(object interface{}) string {
+		h := sha256.New()
+		h.Write([]byte(fmt.Sprintf("%v", resource)))
+		return string(h.Sum(nil))
+	}
```

Comment:
web.Client.Find() can match more than 1 VM depending on how much of the absolute path is specified as part of the name.

Also, for uniqueness, we need to handle the case whereby 2 VMs - one specified by name and the other by ID is actually the same VM.  Duplicate detection becomes a little more complex.